### PR TITLE
#296 - Add InvalidManifestException

### LINKED ISSUE #296: Proper error reporting for manifest validation failures
When manifest is uploaded it undergoes validation, but errors are not properly reported. They should be sent to client in proper fromat with error codes like `MANIFEST_INVALID` and possibly others.

### PR DESCRIPTION
**Pull request by genryxy**
Closes #296 
Added `InvalidManifestException`.

### CODE REVIEW

File: src/main/java/com/artipie/docker/ref/ManifestRef.java [hunk 1]
```diff
@@ -155,7 +156,7 @@ public Key link() {
             } else if (tag.valid()) {
                 ref = new ManifestRef.FromTag(tag);
             } else {
-                throw new IllegalStateException(
+                throw new InvalidManifestException(
```
Comment (olegmoz) [inline src/main/java/com/artipie/docker/ref/ManifestRef.java:159]: @genryxy that is not the case for throwing this error. I am not sure what error should be reported here, but `MANIFEST_INVALID` is for cases whan manifest validation fails, see `AstoManifests.validate` method

**Review by olegmoz — CHANGES_REQUESTED** [review]
@genryxy thanks, please see the comment

**Comment by genryxy** [comment]
@olegmoz thanks, please take a look

**Review by olegmoz — APPROVED** [review]
@genryxy thanks, approved

**Pull request merged**